### PR TITLE
[TensorExpr] Add python bindings.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49698 [TensorExpr] Add python bindings.**

Reincarnation of #47620 by @jamesr66a.

It's just an initial bunch of things that we're exposing to python, more
is expected to come in future. Some things can probably be done better,
but I'm putting this out anyway, since some other people were interested
in using and/or developing this.

Differential Revision: [D25668694](https://our.internmc.facebook.com/intern/diff/D25668694)

### CODE REVIEW

File: test/test_tensorexpr_pybind.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+class kernel_arena_scope(object):
+    def __enter__(self):
+        self.scope = torch._C.te.KernelScope()
+
+    def __exit__(self, typ, val, traceback):
+        self.scope = None
+
+class TestTensorExprPyBind(JitTestCase):
+    def test_simple_sum(self):
+        with kernel_arena_scope():
```
Comment (bertmaher) [inline test/test_tensorexpr_pybind.py:15]: Not a big deal for this PR but omg I want to get rid of KernelArena before we start advertising this API.  Weird low-level memory management hacks shouldn't be leaking up into Python :-p

**Review by bertmaher — COMMENTED** [review]
Is it possible to put the TE-related pybinding code somewhere in jit/tensorexpr instead of in here?  Maybe it's not a big deal it just feels pretty separate from the rest of torch.jit.

**Review by bertmaher — APPROVED** [review]
Looks like a pretty reasonable start to me.  I think we all agree these APIs aren't a desirable end state (I hope) but I'm assuming no one is going to use them externally until we get them into a clean state?

(Particular things that seem clumsy are the limitations on # of args for `Compute`, the fact that we have to use `load` and/or `call` instead of simply indexing, `KernelScope`, `BufferArg`, and the need to create a loopnest, `prepare_for_codegen`, and `simplify` manually)

File: test/test_tensorexpr_pybind.py (hunk 1 above)
Comment (ZolotukhinM) [reply test/test_tensorexpr_pybind.py:15]: 100% agree 😃 
I think even if we keep the arena for now, there is probably a more elegant way of using it for python bindings than I did here.

**Review by ZolotukhinM — COMMENTED** [review]
> Is it possible to put the TE-related pybinding code somewhere in jit/tensorexpr instead of in here? Maybe it's not a big deal it just feels pretty separate from the rest of torch.jit.

Good idea, I moved the TE bindings to a separate file.

> I think we all agree these APIs aren't a desirable end state (I hope) but I'm assuming no one is going to use them externally until we get them into a clean state?

Yes, absolutely. This PR is just to demonstrate the API working end-to-end, making it actually usable was out of the scope :)

**Pull request closed without merging**